### DISCUSSION TITLE
QForm validation error handler with ref errors

### DISCUSSION THREAD

**Discussion by lugrinder** [quasarframework/quasar#16236] [Ideas / Proposals]
I suppose that is not a bug because probably component is not updated at this stage, so I add it as suggestion, but `ref` component passed to `@validation-error` event in QForm doesn't have the `error-message`. If it can't be added to the `ref` component, a second function argument with the error message will help to personalize any user notification at this point, like: `'Hi, review ${ref.label}: ${ref.error-message}'`.

**Comment by yusufkandemir** [comment]
`ref.error-message` is invalid JS, and the props are in camel case in Vue instances when accessing programmatically. So, use `ref.errorMessage` instead.

**Comment by lugrinder** [comment]
That's ok, I've written `error-message` as a reference to the property of the component. But In any case, `ref.errorMessage` and `ref.error`, both are not present in the `ref` component in the `@validation-error` event. `ref.error` is `null` and `ref.errorMessage` is `undefined`, but `ref.hasError` is true. The component shows the error, but at event context, there is no information about the error, only hat has an error.

**Reply by yusufkandemir** [reply]
Your request is clearer now, what you want is not `errorMessage`. You want the error message that is produced as a result of evaluating the `rules` or through the `error`+`errorMessage` combination. That would need to be a new feature.

Possible ways to add this feature:
- Similar to the computed property `hasError`, a new computed property holding the error message can be added.
- `@validation-error` can send the error message as the second parameter.

The first way sounds better to me.
